### DISCUSSION TITLE
Relationship Unique Validation

### DISCUSSION THREAD

**Discussion by iAdil** [Laravel-Backpack/community-forum#579] [Q&A (Help)] [answered]
Hi everyone,
I have an issue and need your help to find a solution.

1. I have two models: **User**(users) and **Code**(codes)
2. User has many codes.

In the **setupUpdateOperation** method I added a field with type: relationship and the subfields.
```
        $this->crud->addFields([
            [
                'name'  => 'codes',
                'label' => 'Codes',
                'type' => 'relationship',
                'subfields'   => [
                    [
                        'name' => 'number',
                        'label' => 'Number',
                        'type' => 'text',
                    ],
                ],
            ]
        ]);
```
In my UserUpdateCrudRequest I have these rules:

```
    public function rules()
    {
        $id = $this->get('id') ?? request()->route('id');

        return [
            'codes.*.number' =>  ['min:3', 'required', "unique:codes,number,$id"],
        ];
    }

```

The issue here is in the **$id** variable, because I'm editing a user object and $id will be related to the editing user.
That's why I will get a validation error during the update, because I can't ignore the correct code by $id

How can I grab the relationship code $id to ignore it inside the unique rule?  

Thank you.

**Comment by christmex** [comment, marked as answer]
Hi @iAdil 

First of all, I'm not experienced to use the relationship field type, but what I got from your question I can say that, we can try these steps
1. get all of the relationship's id
```php
//inside UserUpdateCrudRequest 
$id = $this->get('id') ?? request()->route('id');
$getCodes = User::with('codes')->where('id', $id)->get()->pluck('codes.id', 'id');
```
2. We use Rule Validation inside our Request
```php
//inside UserUpdateCrudRequest , also don't forget to import the Rule Validation Class
use Illuminate\Validation\Rule; // put this

$id = $this->get('id') ?? request()->route('id');
$getCodesId = User::with('codes')->where('id', $id)->get()->pluck('codes.id', 'id');
return [
'codes.*.number' => [
        'min:3',
        'required',
        'distinct',
        Rule::unique('codes', 'number')->where(function ($query) use ($getCodesId) {
            $query->whereNotIn('id', $getCodesId);
            return $query;
        })
    ],
];

```


Reference
https://stackoverflow.com/questions/58520204/how-to-ignore-array-id-for-validating-array-in-laravel
 

Let me know how is it, if you have any issues let me know mate.

Cheers.
